pkg/plugin/resources.go

```diff
@@ -467,6 +469,107 @@ func (a *App) handleLLMState(w http.ResponseWriter, req *http.Request) {
+func getPluginID(ctx context.Context, slug string, grafanaAppURL string, saToken string, gcomAPIKey string) (int, error) {
+	gcomPath := "/api/gnet/instances/" + slug + "/provisioned-plugins/grafana-llm-app"
+	req, err := http.NewRequestWithContext(ctx, "GET", grafanaAppURL+gcomPath, nil)
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", saToken))
+	req.Header.Set("X-Api-Key", gcomAPIKey)
+	if err != nil {
+		return 0, fmt.Errorf("create http request: %w", err)
+	}
+	respBody, err := doRequest(req)
+	if err != nil {
+		return 0, err
+	}
+	plugin := provisionedPlugin{}
+	err = json.Unmarshal(respBody, &plugin)
+	if err != nil {
+		return 0, fmt.Errorf("unmarshal json: %w", err)
+	}
+	return plugin.ID, nil
+}
+
+func (a *App) handleSavePluginSettings(w http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
+		handleError(w, fmt.Errorf("method not allowed: %s", req.Method), http.StatusMethodNotAllowed)
+		return
+	}
+
+	// Read the request body
+	if req.Body == nil {
+		log.DefaultLogger.Warn("Request body is nil")
+		handleError(w, errors.New("request body required"), http.StatusBadRequest)
+		return
+	}
+
+	log.DefaultLogger.Debug("Getting provisioned plugin ID from grafana.com")
+
+	// The HG_INSTANCE_SLUG is only set for Grafana Cloud instances.
+	hgSlug := os.Getenv("HG_INSTANCE_SLUG")
+	var pluginID int
+	var err error
+	if hgSlug != "" && a.settings.EnableGrafanaManagedLLM {
+		pluginID, err = getPluginID(req.Context(), a.settings.Tenant, a.grafanaAppURL, a.saToken, a.settings.GrafanaComAPIKey)
+		if err != nil {
+			handleError(w, fmt.Errorf("get plugin ID: %w", err), http.StatusInternalServerError)
+			return
+		}
+	} else {
+		log.DefaultLogger.Info("Hosted Grafana Slug not found or plugin not provisioned; skipping getting provisioned pluginID")
+		pluginID = 0
+	}
+
+	gcomPath := fmt.Sprintf("/api/gnet/instances/%s/provisioned-plugins/%d", a.settings.Tenant, pluginID)
+	gcomReq, err := http.NewRequestWithContext(req.Context(), "POST", a.grafanaAppURL+gcomPath, req.Body)
+	if err != nil {
+		handleError(w, fmt.Errorf("create gcom request: %w", err), http.StatusInternalServerError)
+		return
+	}
+	gcomReq.Header.Set("Authorization", fmt.Sprintf("Bearer %s", a.saToken))
+	gcomReq.Header.Set("X-Api-Key", a.settings.GrafanaComAPIKey)
+	gcomReq.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	if hgSlug != "" && a.settings.EnableGrafanaManagedLLM {
+		_, err = doRequest(gcomReq)
+	} else {
+		log.DefaultLogger.Info("Hosted Grafana Slug not found or plugin not provisioned; skipping saving settings to grafana.com", "Request details:", fmt.Sprintf(" %+v", gcomReq))
+	}
+
+	// write a success response body since backendSrv.* needs a valid json response body
+	w.WriteHeader(http.StatusOK)
+	_, _ = w.Write([]byte(`{"status": "Success"}`))
+	if err != nil {
+		handleError(w, fmt.Errorf("failed to write response body %w", err), http.StatusInternalServerError)
+		return
+	}
```

Comment:
This block can't really do much after we already write a 200 header so can be removed.